### PR TITLE
(#374) Back To Top

### PR DESCRIPTION
**Pull request by karlikpj**
Closes(#374)

### CODE REVIEW

**Comment by blilianyu** [comment]
Based on the comps, it should be “BACK TO TOP” and not “Return to top” and on mobile the text is swapped out with an arrow.
 
https://publicis.invisionapp.com/console/Back-to-top-cl178a6l13rzm01w5kmj9hrxn/cl178a70u3rzo01w570gchjan/play

File: packages/ncids-css/scss/components/nci-back-to-top/_back-to-top.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  background: color('primary-light');
+  border-top-left-radius: 100%;
+  bottom: 0;
+  box-sizing: border-box;
+  height: 75px;
+	opacity: 0;
+  padding-top: 34px;
+  position: fixed;
+  right: 0;
+	transition: all .25s ease-in-out;
```
Comment (olitharp-nci) [inline packages/ncids-css/scss/components/nci-back-to-top/_back-to-top.scss:12]: can you format this file?

File: packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		console.log('unregister');
+		this.element.classList.remove('show', 'hide', 'nci-footer__return-to-top');
+		this.element.classList.add('grid-container', 'usa-footer__return-to-top');
+	}
+
+	/**
+	 * @private
+	 */
+	private initialize(): void {
+		console.log('int');
```
Comment (olitharp-nci) [inline packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts:61]: clean up console logs

File: packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		const existingComponent = NCIBackToTop._components.get(this.element);
+		if (existingComponent) {
+			existingComponent.unregister();
+		}
+		NCIBackToTop._components.set(this.element, this);
+
+		this.initialize();
+	}
+
+	public static create(element: HTMLElement): NCIBackToTop {
```
Comment (olitharp-nci) [inline packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts:28]: Shouldn't be a component (more like footer collapse than NCI subscribe that the footer uses)

File: docs/content/index.mdx [hunk 4]
```diff
@@ -6,6 +6,7 @@ title: NCI Design System Home
 
 ## Components
 
+- [Back To Top](./components/backtotop)
```
Comment (olitharp-nci) [inline docs/content/index.mdx:9]: remove

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: docs/content/components/footer.mdx [hunk 5]
```diff
@@ -104,7 +104,7 @@ footer.element.addEventListener('usa-footer:nav-links:collapse', (event) => {
 	aria-label="NCI big footer example"
 >
 	<div class="grid-container usa-footer__return-to-top">
-		<a href="#">Return to top</a>
+		<a href="#"><span>BACK TO TOP</span></a>
```
Comment (arcepaul) [inline docs/content/components/footer.mdx:107]: use #top here

File: docs/src/components/footer/nci-big.jsx [hunk 6]
```diff
@@ -10,7 +10,9 @@ export const NCIBigFooter = ({ accountId, categoryId }) => {
 	return (
 		<footer className="usa-footer usa-footer--nci-big">
 			<div className="grid-container usa-footer__return-to-top">
-				<Link to="#top">Return to top</Link>
+				<Link to="#top">
+					<span>BACK TO TOP</span>
```
Comment (arcepaul) [inline docs/src/components/footer/nci-big.jsx:14]: Can we use a text-transform here?  Screen readers don't like all caps and might do strange things

File: docs/content/components/footer.mdx [hunk 7]
```diff
@@ -104,7 +104,7 @@ footer.element.addEventListener('usa-footer:nav-links:collapse', (event) => {
 	aria-label="NCI big footer example"
 >
 	<div class="grid-container usa-footer__return-to-top">
-		<a href="#">Return to top</a>
+		<a href="#top"><span>Back To Top</span></a>
```
Comment (arcepaul) [inline docs/content/components/footer.mdx:107]: aria-label missing here

**Comment by blilianyu** [comment]
@karlikpj Can we make that whole area clickable? Or we can't because of USWDS?

**Review by blilianyu — APPROVED** [review]

File: packages/ncids-css/scss/components/nci-back-to-top/_back-to-top.scss [hunk 8]
```diff
@@ -0,0 +1,73 @@
+.nci-footer__return-to-top {
+  bottom: 0;
+  box-sizing: border-box;
+  height: 75px;
+  opacity: 0;
+
+  position: fixed;
+  right: 0;
+  transition: all 0.25s ease-in-out;
+	text-transform: uppercase;
```
Comment (olitharp-nci) [inline packages/ncids-css/scss/components/nci-back-to-top/_back-to-top.scss:10]: can you reformat this file?

File: packages/ncids-css/scss/components/nci-back-to-top/_back-to-top.scss [hunk 9]
```diff
@@ -0,0 +1,73 @@
+.nci-footer__return-to-top {
```
Comment (olitharp-nci) [inline packages/ncids-css/scss/components/nci-back-to-top/_back-to-top.scss:1]: we don't have a nci-footer, so probably should be:

```suggestion
.usa-footer__nci-return-to-top {
```

File: packages/ncids-css/scss/components/usa-footer/_nci-big.scss [hunk 10]
```diff
@@ -12,3 +12,4 @@
 // NCI Component
 @import 'internal/nci-big';
 @import 'internal/print';
+@import '../nci-back-to-top/back-to-top';
```
Comment (olitharp-nci) [inline packages/ncids-css/scss/components/usa-footer/_nci-big.scss:15]: add new line to end of file

File: packages/ncids-js/src/components/usa-footer/__tests__/nci-footer-back-to-top.test.ts [hunk 11]
```diff
@@ -0,0 +1,66 @@
+import '@testing-library/jest-dom';
+import '@testing-library/jest-dom/extend-expect';
+
+import { fireEvent, waitFor } from '@testing-library/dom';
+//import userEvent from '@testing-library/user-event';
```
Comment (olitharp-nci) [inline packages/ncids-js/src/components/usa-footer/__tests__/nci-footer-back-to-top.test.ts:5]: remove commented out code

File: packages/ncids-js/src/components/usa-footer/nci-big-footer.component.ts [hunk 12]
```diff
@@ -1,7 +1,7 @@
 import { NCIBigFooterOptions } from './nci-big-footer-options';
 import { FooterCollapse } from './utils/footer-collapse';
+import { NCIBackToTop } from './utils/footer-back-to-top.component';
```
Comment (olitharp-nci) [inline packages/ncids-js/src/components/usa-footer/nci-big-footer.component.ts:3]: file should just be 

```suggestion
import { NCIBackToTop } from './utils/footer-back-to-top';
```

File: packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+			this.element.classList.add('hide');
+		}
+	}
+
+	/**
+	 * Resets component to a clean state.
+	 * @public
+	 */
+	public unregister(): void {
+		this.element.classList.remove('show', 'hide', 'nci-footer__return-to-top');
```
Comment (olitharp-nci) [inline packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts:31]: see note in scss file about naming

```suggestion
		this.element.classList.remove('show', 'hide', 'usa-footer__nci-return-to-top');
```

File: packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+	/**
+	 * @private
+	 */
+	private initialize(): void {
+		this.element.classList.remove(
+			'grid-container',
+			'usa-footer__return-to-top'
+		);
+		this.element.classList.add('nci-footer__return-to-top', 'hide');
```
Comment (olitharp-nci) [inline packages/ncids-js/src/components/usa-footer/utils/footer-back-to-top.component.ts:43]: here too

File: testing/ncids-css-testing/stories/components/usa-footer/usa-footer.stories.jsx [hunk 15]
```diff
@@ -6,3 +6,5 @@ export default {
 export * from './nci-big.stories';
 export * from './nci-big.initial.stories';
 export * from './nci-big.sign-up-error.stories';
+export * from './nci-big.back-to-top.stories';
+export * from './nci-big.back-to-top-es.stories';
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/usa-footer/usa-footer.stories.jsx:10]: new line at end of file

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Pull request merged**